### PR TITLE
Update test matrix to use 8.15.3

### PR DESCRIPTION
**Pull request by dimuon**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ jobs:
           - '8.12.2'
           - '8.13.4'
           - '8.14.3'
-          - '8.15.0'
+          - '8.15.3'
```
Comment (biscout42) [inline .github/workflows/test.yml:126]: @dimuon , IIRC we would need to add this version to synthetic install workaround here https://github.com/elastic/terraform-provider-elasticstack/blob/main/.github/workflows/test.yml#L165

Comment (dimuon) [reply .github/workflows/test.yml:126]: good catch 👍 thanks!

Comment (biscout42) [reply .github/workflows/test.yml:126]: Hm... force install synthetics build step shows for `8.15.3`:
> Kibana server is not ready yet

I try to re-run it 🤔

**Comment by dimuon** [comment]
There are few failing tests for 8.15.3.

**Comment by biscout42** [comment]
@dimuon , I think we also need to update link 156 to have fleet setup
```
        if: matrix.version == '8.10.3' || matrix.version == '8.11.4' || matrix.version == '8.12.2' || matrix.version == '8.13.4' || matrix.version == '8.14.3' || matrix.version == '8.15.0'
```

I'm sorry, somehow I can't make a suggestion in your PR at this line 🤔

**Comment by biscout42** [comment]
Thanks, it worked 🎉

**Comment by dimuon** [comment]
@biscout42 , thanks for help!

**Comment by dimuon** [comment]
@biscout42 can you please approve it?

**Review by biscout42 — APPROVED** [review]

**Pull request merged**